### DISCUSSION TITLE
How to get the version history storage of all documents in a library?

### DISCUSSION THREAD

**Discussion by khha** [pnp/powershell#4626] [Q&A] [answered]
How to get the version history storage of all documents in a library? I try below, but it is not working.

```
$listItems = Get-PnPFolderItem -Identity $library.RootFolder -Recursive -ItemType File

ForEach($item in $listItems) {
                                $file = Get-PnPProperty -ClientObject $item -Property file
                                $versions = Get-PnPProperty -ClientObject $file -Property versions
                                $fileSizeinKB = [Math]::Round(($item.FieldValues.File_x0020_Size/1KB),2)
}
```

I got error:

Get-PnPProperty: 'file' is not a member of type 'Microsoft.SharePoint.Client.File' (Parameter 'propertyOrFieldName')

**Comment by ToddKlindt** [comment, marked as answer]
There are a few examples in the [Script Samples](https://pnp.github.io/script-samples/?query=version). Maybe one of them can nudge you in the right direction.
